bf2fc6cc711aee1a0c2a/kas-fleet-manager
# test: fix and refactor kafka integration tests to work with rhsso

## Description
Refactor and fix any issues found when running integration tests with both MAS SSO and RHSSO clients. 

Please note that when `redhat_sso` is set as the selected SSO provider, this will call to the actual sso instance whether the environment selected is integration or development. Because of this, we will need to get an actual token from rhsso within our tests. If mas_sso is selected, we can continue on using the mock token. 

There are also a few differences in the responses between these two clients. Any assertions that has different results between the two have been commented out for now.

Note that this PR only fixes kafka related tests

## Verification Steps
1. Ensure CI is passing (Runs the tests with `OCM_ENV=integration` AND `SelectSSOProvider=MAS_SSO ` configuration)
2. Run the kafka test with the following configurations:
    * `OCM_ENV=integration` and `SelectSSOProvider: REDHAT_SSO`
    * `OCM_ENV=development` and `SelectSSOProvider: MAS_SSO`
       * See results here: [mas_sso_dev_env_test_results](https://gist.github.com/JameelB/afd2aa3d439765d36b3f634148cf804b#file-mas_sso_dev_env_test_results)
    * `OCM_ENV=development` and `SelectSSOProvider: REDHAT_SSO`
      * See results here: [rhsso_dev_env_test_results](https://gist.github.com/JameelB/afd2aa3d439765d36b3f634148cf804b#file-rhsso_dev_env_test_results)
      * Note that connector tests are still failing when using rhsso provider. This will be fixed in a separate PR

When running with 'redhat_sso' + `OCM_ENV=integration`, please update any references of `sso.redhat.com` to `sso.qa.redhat.com` in the [keycloakConfig](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/config.go#L89) and [serverConfig](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/server/server_config.go). Do not modify the mas-sso-base-url as this is still use for configuration of osd idp. A valid rhsso service account client id and secret is required to run this, please message me for that information when verifying these changes.

## Checklist (Definition of Done)
